src/libraries/Native/Unix/System.Globalization.Native/pal_icushim_static.c

```diff
@@ -146,7 +154,7 @@ int32_t GlobalizationNative_LoadICUData(const char* path)
 
     fclose(fp);
 
-    if (load_icu_data(icu_data) == 0) {
+    if (load_icu_data(icu_data, strcasecmp("icudt.dat", path)) == 0) {
```

Comment:
Will this always be named icudt.dat? I think in some cases the platform native .dat file might have a version number in its path.

Reply:
should rename all the shards to something like .app.dat and assume anything that doesn't match that pattern should use setCommonData?